File: ResearchCruiseApp-API/ResearchCruiseApp-API/Application/Models/DTOs/Forms/FormAInitValuesDto.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;
using ResearchCruiseApp_API.Application.Models.DTOs.CruiseApplications;

namespace ResearchCruiseApp_API.Application.Models.DTOs.Forms;


[JsonObject(NamingStrategyType = typeof (CamelCaseNamingStrategy))]
public class FormAInitValuesDto
{
    public List<FormUserDto> CruiseManagers { get; set; } = [];
    public List<FormUserDto> DeputyManagers { get; set; } = [];
    public List<string> Years { get; set; } = [];
    public List<string> ShipUsages { get; set; } = [];
    public List<ResearchAreaDto> ResearchAreas { get; set; } = [];
    public List<string> CruiseGoals { get; set; } = [];
    public List<ResearchTaskDto> HistoricalTasks { get; set; } = [];
    public List<UgUnitDto> UgUnits { get; set; } = [];
}

File: frontend/src/features/cruise-applications/models/FormAInitValuesDto.ts
import { ContractDto } from '@/features/cruise-applications/models/ContractDto';
import { FormUserDto } from '@/features/cruise-applications/models/FormUserDto';
import { PublicationDto } from '@/features/cruise-applications/models/PublicationDto';
import { ResearchAreaDto } from '@/features/cruise-applications/models/ResearchAreaDto';
import { ResearchTaskDto } from '@/features/cruise-applications/models/ResearchTaskDto';
import { SpubTaskDto } from '@/features/cruise-applications/models/SpubTaskDto';
import { UgUnitDto } from '@/features/cruise-applications/models/UgUnitDto';

export type FormAInitValuesDto = {
  cruiseManagers: FormUserDto[];
  deputyManagers: FormUserDto[];
  years: string[];
  shipUsages: string[];
  standardSpubTasks: string[];
  researchAreas: ResearchAreaDto[];
  cruiseGoals: string[];
  historicalResearchTasks: ResearchTaskDto[];
  historicalContracts: ContractDto[];
  ugUnits: UgUnitDto[];
  historicalGuestInstitutions: string[];
  historicalSpubTasks: SpubTaskDto[];
  historicalPublications: PublicationDto[];
};
